### ISSUE TITLE
Make sure Cloudflare doesn’t cache documentation 404s

### ISSUE THREAD

**Issue by heckj** [SwiftPackageIndex/SwiftPackageIndex-Server#1787]
**Please describe the bug**

I added the .spi.yml and pushed, and the documentation link now shows for the package [heckj/Lindenmayer](https://swiftpackageindex.com/heckj/Lindenmayer/), but the backing documentation hasn't appeared, instead there's an [error page](https://swiftpackageindex.com/heckj/Lindenmayer/main/documentation/lindenmayer)

**Explain the steps needed to reproduce the bug**

1. Added .spi.yml, which has *two* targets listed (Lindenmayer and LindenmayerViews), pushed code
2. forgot to add the docc plugin, so added that and re-pushed code
3. Waited 18+ hours before submitting the bug report ;-)

.spi.yml file included in repo:
```yaml
version: 1
builder:
  configs:
    - documentation_targets: [Lindenmayer, LindenmayerViews]
```

When I generated the docs myself, there wasn't much of an option to include both in the Github Pages, so I defaulted to just Lindenmayer (https://heckj.github.io/Lindenmayer/documentation/lindenmayer/) - so I'm sure the doc generation is working

**Comment by finestructure** [comment]
Hi Joe, both docs sets are there.

- https://swiftpackageindex.com/heckj/Lindenmayer/main/documentation/lindenmayerviews works
- https://swiftpackageindex.com/heckj/Lindenmayer/main/documentation/lindenmayer doesn't but
- https://swiftpackageindex.com/heckj/lindenmayer/main/documentation/lindenmayer does (note the lowercase "L")

I wonder if Cloudflare is somehow holding on to our error page? Although that should be long expired, it's only 3h. Unfortunately I can't purge it to reset, so we'll have to check back in a bit to see if it shakes itself loose.

**Comment by finestructure** [comment]
And there it is! We need to check why CF is holding on to the error page. I think unless we exclude that from caching it’ll extend the update delay unnecessarily by 3h.

Thanks again for reporting this, I’ll keep this issue open to track the CF change.

**Comment by heckj** [comment]
Thanks!

**Comment by finestructure** [comment]
This would also be addressed by #1784 , because we'd only offer a link once we know there's content in the bucket. (There'd be still some ways this could 404 for odd reasons so we'd want to avoid caching those regardless.)
